server/pulp/server/content/web/views.py

```diff
@@ -68,7 +69,13 @@ def x_send(path):
         :rtype: django.http.HttpResponse
         """
         if os.access(path, os.R_OK):
-            reply = HttpResponse()
+            content_type = mimetypes.guess_type(path)
```

Comment:
We could use a library like libmagic to determine the file type (which will be much more accurate), but I was hesitant to introduce a new dependency. The downside to this method is that if the file suffix is wrong/missing (like the `listing` files which are text of some variety) the 'application/octet-stream' is used.